executor/native_linux_executor.cpp

```diff
@@ -485,9 +487,14 @@ void NativeLinuxExecutor::Run(const u8 *buf, u32 len, u32 timeout_ms) {
 		// stdin is bound to /dev/null ( Providing "nothing" as input ).
                 dup2(null_fd, 0);
             }
+	    std::vector< char* > temporal_envp;
             // Execute new executable binary on the child process.
 	    // If failed, that matter is recorded to child_state.
-            child_state->exec_result = execv(cargv[0], (char**)cargv.data());
+            child_state->exec_result = execve(
+                cargv[0],
+                (char**)cargv.data(),
+                GetJoinedEnvironmentVariables( temporal_envp )
+	    );
```

Comment:
While calling `GetJoinedEnvironmentVariables` here is totally valid, I think it is a bit naive to create a `std::vector<char*>` instance every time.

How about make NativeLinuxExecutor have `std::vector<char*> temporal_envp` as a private member variable like `NativeLinuxExecutor::cenvp`? It's just the same as `Executor::cargv`. Because the lifetime of `cenvp` would be the same as that of `environment_variables`, I don't see it so problematic.

Reply:
fixed